lib/runner.go

```diff
@@ -99,3 +99,8 @@ type Summary struct {
 	NoColor         bool          // TODO: drop this when noColor is part of the (runtime) options
 	UIState         UIState
 }
+
+// String Overiding for tests
+func (s *Summary) String() string {
+	return "&{Summary:map[...]...}"
+}
```

Comment:
I guess when we agree on the generation part then we should update this with some additional information. We might have `Summary{Metrics: number, RootGroup: name/(or nil), and the other fields}`.